### PR TITLE
Config tests

### PR DESCRIPTION
**Pull request by cherednichenko**
I did the following:
1 - created tests for MapConfig, QueueConfig, CacheConfig (validation for BackupCount and AsyncBackupCount).
2 - added fix for QueueConfig.java.
3 - did some small updates/refactoring.

All tests are passed.

Only one worry I have - I didn't remove anything related to this file 'hazelcast/src/test/java/com/hazelcast/config/MapConfigTest.java' but GitHub shows as I removed all content that's not true (maybe the file was moved BUT I checked on hazelcast and it's still there). 

Thanks a lot.

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/config/QueueConfigTest.java [hunk 1]
```diff
@@ -60,5 +60,39 @@ public void testReadOnlySetName() {
...
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void setAsyncBackupCount_whenTooLarge(){
+        QueueConfig config = new QueueConfig();
+        config.setAsyncBackupCount(200); //max allowed is 6..
+    }
 
+    @Test(expected = IllegalArgumentException.class)
+    public void setBackupCount_tooSmall(){
```
Comment (bilalyasar) [inline hazelcast/src/test/java/com/hazelcast/config/QueueConfigTest.java]: i guess `setBackupCount_whenNegative`makes more sense :)

**Comment by bilalyasar** [comment]
it seems ok to me, but can you fix checkstyle and put back removed MapConfigTest.java file :)

**Comment by cherednichenko** [comment]
@bilalyasar 
Ok, I'll fix checkstyle - actually nobody showed me that procedure before :)

MapConfigTest.java file actually wasn't removed, check it here - https://github.com/cherednichenko/hazelcast/blob/configTests/hazelcast/src/test/java/com/hazelcast/config/MapConfigTest.java#L25 . At least - I didn't remove it :) 
I see here two options:
1 - somebody committed the same file or replicate it.
2 - it's some bug related to Git or IDEA.

File: hazelcast/src/test/java/com/hazelcast/config/MapConfigTest.java [hunk 2]
```diff
@@ -1,238 +1 @@
-/*
```
Comment (pveentjer) [inline hazelcast/src/test/java/com/hazelcast/config/MapConfigTest.java]: What happened to MapConfigTest? You deleted it?

**Comment by pveentjer** [comment]
@hasancelik can you help @cherednichenko out to clean up this PR and see what happened to the MapConfigTest? I would like to get this pr moving again.

File: hazelcast/src/test/java/com/hazelcast/config/MapConfigTest.java (hunk 2 above)
Comment (cherednichenko) [inline hazelcast/src/test/java/com/hazelcast/config/MapConfigTest.java]: I didn't actually - I mean that I didn't delete anything but I'll try to figure out - what's happened exactly.

Comment (cherednichenko) [inline hazelcast/src/test/java/com/hazelcast/config/MapConfigTest.java]: Actually with check-style parameter build is failing but this file was changed before me and it was excluded from jacoco-coverage -- https://github.com/hazelcast/hazelcast/commit/4585c99f97c2719f6db46d5c3528d3714e301a40 

Maybe it's a reason why it's failing but the question is - should I change this file or not?!

**Comment by cherednichenko** [comment]
New PR #4988 was created.
This one is closed for now.

**Pull request closed without merging**